src/lightbox/LightboxContainer.js

```diff
@@ -0,0 +1,140 @@
+/* flow */
+import React from 'react';
+import { View, StyleSheet, Dimensions, Easing } from 'react-native';
+import { connect } from 'react-redux';
+import PhotoView from 'react-native-photo-view';
+import { connectActionSheet } from '@expo/react-native-action-sheet';
+
+import Header from './LightboxHeader';
+import Footer from './LightboxFooter';
+import boundActions from '../boundActions';
+import { constructActionSheetButtons, executeActionSheetAction } from './LightboxActionSheet';
+import { NAVBAR_HEIGHT, LIGHTBOX_FOOTER_OFFSET } from '../styles';
+
+const WINDOW_WIDTH = Dimensions.get('window').width;
+const WINDOW_HEIGHT = Dimensions.get('window').height;
+const FOOTER_HEIGHT = 44;
+
+const styles = StyleSheet.create({
+  img: {
+    width: WINDOW_WIDTH,
+    height: 300,
+    flex: 1,
+  },
+  header: {
+    width: WINDOW_WIDTH,
+    height: NAVBAR_HEIGHT,
+    paddingLeft: 15,
+    paddingRight: 5,
+  },
+  footer: {
+    width: WINDOW_WIDTH,
+    height: FOOTER_HEIGHT,
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    paddingLeft: 15,
+    paddingRight: 5,
+  },
+  overlay: {
+    backgroundColor: 'rgba(0, 0, 0, 0.2)',
+    position: 'absolute',
+    zIndex: 1,
+  },
+  container: {
+    flex: 1,
+    justifyContent: 'space-between',
+    alignItems: 'center',
+  },
+});
+
+export default connectActionSheet(
+  connect(null, boundActions)(
+    class LightboxContainer extends React.Component {
+      state = {
+        movement: 'out',
+      };
+
+      handleImagePress = () => {
+        this.setState({
+          movement: this.state.movement === 'out' ? 'in' : 'out',
```

Comment:
When a state depends on its previous value use the setState(function) variation (see the help, it is not a well known feature)